### PR TITLE
add cask for hab

### PR DESCRIPTION
**Pull request by cullenmcdermott**
#### Adding a new cask
- [✅] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [✅] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [✅] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [✅] Commit message includes cask’s name.
- [✅] `brew cask audit --download {{cask_file}}` is error-free.
- [✅] `brew cask style --fix {{cask_file}}` left no offenses.
- [✅] `brew cask install {{cask_file}}` worked successfully.
- [✅] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you for the contribution. Seems like this is an open-source CLI tool. In that case, and to [avoid duplication](https://github.com/caskroom/homebrew-cask/issues/15603) this should be only in homebrew.

I saw https://github.com/Homebrew/homebrew-core/pull/1994, but this should be reworked as a formula that is compiled, rather than a downloadable cask.

Pinging @DomT4.

**Comment by cullenmcdermott** [comment]
Yeah that was my initial plan but Habitat has some dependencies on on the casks vmware-fusion and dockertoolbox in order to compile it from source and I wasn't able to find a good way to depend on a cask from inside of a formula. I'll look into it some more.

**Comment by DomT4** [comment]
@vitorgalvao Hey Vitor. My message about the Cask accepting binaries is a generic reply, to be honest. I'm happy to poke people towards `homebrew/binary` instead if Cask wants to completely avoid CLI stuff going forwards?

If it can build from source & someone's willing to work on that we're certainly happy to take it in `homebrew/core`, but I wasn't sure if that was the case here.

**Comment by vitorgalvao** [comment]
> I'm happy to poke people towards `homebrew/binary` instead if Cask wants to completely avoid CLI stuff going forwards?

We’re happy to take CLI stuff, it just seemed to me that since this is open-source and CLI-only, homebrew-core would be the better place.

> but Habitat has some dependencies on on the casks vmware-fusion and dockertoolbox in order to compile it from source and I wasn't able to find a good way to depend on a cask from inside of a formula.

If there’s such a hurdle in getting this to compile from source, then sure, we’ll take it here.

> If it can build from source & someone's willing to work on that we're certainly happy to take it in `homebrew/core`, but I wasn't sure if that was the case here.

If that happens, then we can switch (remove from HBC, add to HB).

File: Casks/hab.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask 'hab' do
+  version '0.6.0-20160613151520'
+  sha256 '2251891c376bd3786a2c8242006367a3fd9f896cab850472c73886c6556598d1'
+
+  # bintray is where the habitat binaries are stored
```
Comment (vitorgalvao) [inline Casks/hab.rb]: Comment needs to be reworked. Can you show me where on the official websites do they link to this URL? I need to verify it.

File: Casks/hab.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  url "https://api.bintray.com/content/habitat/stable/darwin/x86_64/hab-#{version}-x86_64-darwin.zip?bt_package=hab-x86_64-darwin"
```
Comment (vitorgalvao) [inline Casks/hab.rb]: Can’t the `?bt_package=hab-x86_64-darwin` part be removed?

File: Casks/hab.rb (hunk 1 above)
Comment (cullenmcdermott) [inline Casks/hab.rb]: Here is the page where they link to it: https://www.habitat.sh/docs/get-habitat/

How should it be reworded?

File: Casks/hab.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cask 'hab' do
+  version '0.7.0-20160614231131'
+  sha256 '93fac880261df34c75ac5775c9febb536b82ac955cd6df873bbfec1cf697ed50'
+
+  # bintray is where the habitat binaries are stored
```
Comment (vitorgalvao) [inline Casks/hab.rb]: Should be `# habitat.bintray.com was verified as official when first introduced to the cask` ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

**Pull request merged**